src/main/kotlin/withdrawalservice/WithdrawalServiceImpl.kt

```diff
@@ -41,15 +43,23 @@ data class RollbackApproval(
  */
 class WithdrawalServiceImpl(
     val withdrawalServiceConfig: WithdrawalServiceConfig,
+    val keypair: Keypair,
+    val irohaNetwork: IrohaNetwork,
     private val irohaHandler: Observable<SideChainEvent.IrohaEvent>
 ) : WithdrawalService {
     private val notaryPeerListProvider = NotaryPeerListProviderImpl()
     private val tokensProvider: EthTokensProvider = EthTokensProviderImpl(withdrawalServiceConfig.db)
     private val masterAccount = withdrawalServiceConfig.notaryIrohaAccount
 
     private fun findInAccDetail(acc: String, name: String): String {
-        val relays = getRelays(withdrawalServiceConfig.iroha, acc, withdrawalServiceConfig.registrationIrohaAccount)
-        for (record in relays) {
+        val relays = getRelays(
+            withdrawalServiceConfig.iroha,
+            keypair,
+            irohaNetwork,
+            acc,
+            withdrawalServiceConfig.registrationIrohaAccount
+        )
+        for (record in relays.get()) {
```

Comment:
May be we should check was `getRelays()` call succeeded? Probably `findInAccDetail` can also return Result.